### ISSUE TITLE
Request: Printable Directions With Highlighted Route

### ISSUE THREAD

**Issue by PorcelainMouse** [openstreetmap/openstreetmap-website#1273]
After using the directions or routing feature of OSM, directions are displayed over the map on the left side.  However, when printing such a web-page (in Firefox, at least), the directions are missing and only the map is printed.  Please add some way to make a hard copy of the directions and the map.  It doesn't need to look exactly as displayed, if that is more difficult.

**Comment by mmd-osm** [comment]
See [Help OSM](https://help.openstreetmap.org/questions/51776/how-to-print-driving-directions) for full discussion.

**Comment by CodeMouse92** [comment]
Agreed, this is the only thing preventing me from switching to OSM!

**Comment by michaelblyons** [comment]
Seems like an odd oversight to affirmatively remove the directions list from the `print` stylesheet. 😕

**Comment by tomhughes** [comment]
Do you have any evidence that is what is happening?

I suspect it's more likely that surrounding chrome (like the sidebar) was removed in order to include as much of the map as possible and that has the side effect of removing routing information.

That said, it is explicitly not the goal of openstreetmap.org to provide a full service google maps replacement for end users, so being able to print directions to take with you on a journey or whatever is not really in scope as a goal.

**Comment by michaelblyons** [comment]
It's the whole sidebar, yes.
```css
header, #sidebar, #permalink, .leaflet-control {
  display: none;
}
```

**Comment by landonjefftaylor** [comment]
I fully support this request. I drive frequently in rural areas without cell signal and it's useful to have printed driving directions. Screen snipping works okay, but it would be so nice to have a decent print feature like other map tools have.

**Comment by gravitystorm** [comment]
This is mostly fixed by #4184. 

However, there are several features that are not implemented, or that could be improved, so it's not a complete solution. I hope that it is sufficient for most people's needs, since we continue to focus our efforts on other priorities.
